### PR TITLE
Add Msgpack codec

### PR DESCRIPTION
**Pull request by pschwartz**

### CODE REVIEW

File: codec/msgpack/msgpack_test.go [hunk 1]
```diff
@@ -0,0 +1,11 @@
+package msgpack
+
+import (
+	"testing"
+
+	"github.com/asdine/storm/codec/internal"
+)
+
+func TestMSGPACK(t *testing.T) {
```
Comment (bep) [inline codec/msgpack/msgpack_test.go]: I believe the correct test name would be `TestMsgpack`

File: codec/msgpack/msgpack.go [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+func (m msgpackCodec) Marshal(v interface{}) ([]byte, error) {
+	return mp.Marshal(v)
+}
+
+func (m msgpackCodec) Unmarshal(b []byte, v interface{}) error {
+	return mp.Unmarshal(b, v)
+}
+
+func (m msgpackCodec) Name() string {
```
Comment (bep) [inline codec/msgpack/msgpack.go]: What is the point ot the `Name` func? It isn't (as I'm aware of) an interface method.

File: codec/msgpack/msgpack_test.go (hunk 1 above)
Comment (pschwartz) [inline codec/msgpack/msgpack_test.go]: Fixed

File: codec/msgpack/msgpack.go (hunk 2 above)
Comment (pschwartz) [inline codec/msgpack/msgpack.go]: This was to match the other codec's. It is not part of the interface, but all include it.

Comment (pschwartz) [inline codec/msgpack/msgpack.go]: And actually, it is part of the interface in codec/codec.go.

Comment (bep) [inline codec/msgpack/msgpack.go]: Ah, OK -- I created this package, but I guess stuff have happened since then.

Comment (bep) [inline codec/msgpack/msgpack.go]: I can merge this if you can fix your commit message:

1. Remove the ad line in the title 
2. Remove the Signed-Off-By line in the body

Comment (pschwartz) [inline codec/msgpack/msgpack.go]: done

Comment (bep) [inline codec/msgpack/msgpack.go]: It still says "compact and fast data storage format", which is what I was referring to as "ad line". I may be pedantic about commit messages, but they should be neutral, so the correct one in this case is "Add msgpack codec". Simple and to the point.

Comment (pschwartz) [inline codec/msgpack/msgpack.go]: Done

**Pull request merged**